### PR TITLE
Adding sync scenarios to public preview swagger

### PR DESCRIPTION
**Pull request by ash1625**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Tasks.json [hunk 1]
```diff
@@ -219,6 +243,30 @@
           "$ref": "#/definitions/ProjectTaskProperties"
         }
       ]
+    },
+	"ConnectToTargetAzureDbForPostgreSqlSyncTaskProperties": {
+      "x-ms-discriminator-value": "ConnectToTarget.AzureDbForPostgreSql.Sync",
+      "type": "object",
+      "description": "Properties for the task that validates connection to Azure Db For PostgreSql server and target server requirements for online migration",
```
Comment (hitenjava) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Tasks.json]: "Azure Database for PostgreSQL"

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Data Migration Service Resource Provider",
```
Comment (hitenjava) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json]: "Azure Database Migration Service..." - here and in other files.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json [hunk 3]
```diff
@@ -96,6 +96,59 @@
...
+            "$ref": "./Common.json#/definitions/ReportableException"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "MigrateMISyncCompleteCommandProperties": {
+      "x-ms-discriminator-value": "Migrate.SqlServer.AzureDbSqlMi.Complete",
+      "type": "object",
+      "description": "Properties for the command that completes sync migration for a managed instance database.",
```
Comment (hitenjava) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json]: "Azure SQL Database Managed Instance"

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json [hunk 4]
```diff
@@ -96,6 +96,59 @@
...
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/CommandProperties"
+        }
+      ]
+    },
+    "MigrateMISyncCompleteCommandInput": {
+      "type": "object",
+      "description": "Input for command that completes sync migration for a managed instance database.",
```
Comment (hitenjava) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json]: Azure SQL Database Managed Instance

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json (hunk 3 above)
Comment (hitenjava) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json]: "online migration"

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json (hunk 4 above)
Comment (hitenjava) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json]: "online migration"

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json (hunk 2 above)
Comment (neetusingh-ms) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json]: This is in other existing files as well. Let's keep it as is for this change and may be add a task to change it across all the files later.

Comment (hitenjava) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json]: I thought we had already updated it for the public repo ones

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    "version": "2017-11-15-privatepreview"
```
Comment (neetusingh-ms) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json]: change version here and other files.

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json (hunk 3 above)
Comment (ash1625) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/Commands.json]: I think we are removing online and keeping sync everywhere

File: specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json (hunk 2 above)
Comment (ash1625) [reply specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/ConnectToSourcePostgreSqlSyncTask.json]: Already changed in all the files

**Review by neetusingh-ms — APPROVED** [review]
Thanks Ashish.

**Review by hitenjava — APPROVED** [review]

**Comment by praries880** [comment]
@ravbhatnagar can you kindly take a look at this?

File: specification/datamigration/resource-manager/Microsoft.DataMigration/stable/2018-04-19/definitions/ConnectToTargetSqlSqlDbSyncTask.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{
+  "swagger": "2.0",
```
Comment (neetusingh-ms) [inline specification/datamigration/resource-manager/Microsoft.DataMigration/stable/2018-04-19/definitions/ConnectToTargetSqlSqlDbSyncTask.json]: name of the has SqlSql..

**Comment by ravbhatnagar** [comment]
@ash1625 - There are a lot of changes in this PR. It might be quicker to review this over a meeting. Please feel to set something up with GauravBh and armapireview.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side. Noted few issues during the in-person review. Service team is close to GA and may not be able to likely fix it for this api-version. But will likely consider for the next one.
1. Rename "errors" in MigrateSyncCompleteCommandOutput properties.
2. How to get information of all the commands on a task.
3. Implementation for updating an existing tasks.
4. migrationSetting/sourceSetting/targetSetting object (expand this object for known properties ) good for CLI/powershell users.

**Review by praries880 — APPROVED** [review]

**Pull request merged**